stable/redis-ha/templates/redis-ha-announce-service.yaml

```diff
@@ -0,0 +1,33 @@
+{{- $fullName := include "redis-ha.fullname" . }}
+{{- $replicas := int .Values.replicas }}
+{{- $root := . }}
+{{- range $i := until $replicas }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ $fullName }}-announce-{{ $i }}
```

Comment:
I think maybe these services are not needed, because `StatefulSet` can have Stable Network ID through headless service.
https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#stable-network-id

Reply:
@ceshihao This is how the chart is currently using for discovery however the problem comes when a pod gets recreated and is assigned a new IP address (but same name). This causes issues because Redis itself resolves the name and updates the config with only the IP. This causes issues when a network partition occurs and the pod is isolated. With having a service for each pod, it provides a stable IP that the Quorum can always reference and recover missing redis/sentinel even after network partitions. 

These really are only necessary for stable inter pod discovery and all clients should still use the `Headless Service`